app/org/sagebionetworks/bridge/upload/UploadValidationTask.java

```diff
@@ -111,6 +124,30 @@ public void run() {
         }
 
         // TODO: if validation fails, wipe the files from S3
+
+        // dedupe logic over here:
+        dedupeHelper(context.getRecordId());
+    }
+
+    /**
+     * helper method: query healthdatarecord by healthcode, schema and createdOn from the upload just finished above,
+     * if there are more than 1 such records in ddb, there are duplicates. log that information but not stop or delete anything right now
+     * @param healthRecordId
+     */
+    private void dedupeHelper(String healthRecordId) {
+        // get necessary information for query first
+        HealthDataRecord record = healthDataService.getRecordById(healthRecordId);
+
+        Long createdOn = record.getCreatedOn();
+        String healthCode = record.getHealthCode();
+        String schemaId = record.getSchemaId();
+
+        List<HealthDataRecord> retList = healthDataService.getRecordsByHealthcodeCreatedOnSchemaId(healthCode, createdOn, schemaId);
+
+        if (retList.size() > 1) {
+            logger.warn(String.format("Duplicate health data records for record id: %s, health code: %s, created on:  %s, schema id: %s, duplicate size: %s",
+                    healthRecordId, healthCode, createdOn, schemaId, retList.size()));
+        }
```

Comment:
It's not enough to simply log that a duplicate exists. If we want to analyze, we need to actually identify the duplicates.

Also, this should be info, not warning. Warnings mean there's an immediately actionable if there are lots of them. This doesn't qualify.

Lastly, you'll want a unit test to make sure this works. It might be worth factoring the warning message to a "handleDuplicate()" method that you can spy and verify that it was called. You'll also want to test the cases where there is no record for this upload, where the record is not a duplicate, where there is a duplicate, and where there's more than one duplicate.